packages/commonwealth/server/util/roles.ts

```diff
@@ -0,0 +1,262 @@
+import { Model, Transaction, Op, FindOptions } from 'sequelize';
+import { DB } from 'server/models';
+import { sequelize } from '../database';
+import { Permission } from '../models/role';
+import { RoleAssignmentAttributes } from '../models/role_assignment';
+
+export class RoleInstanceWithPermission {
```

Comment:
Why is the interface to this so different than the one in the spec?

Reply:
We discussed this in chat, but we cannot override the type, so we had to build a class instead. I should have updated the spec, that is my bad!

Reply:
can refactor this down the line, but works as is!